src/kili/entrypoints/mutations/project/import_csv.py

```diff
@@ -0,0 +1,141 @@
+    csv_file: str,
+    content_column: str,
+    unique_categories: Optional[List[str]],
+    category_column: Optional[str],
+    external_id_column: Optional[str],
+    csv_separator: str,
+    label_type: LabelType,
+    project_id: Optional[str],
+    project_title: Optional[str],
+    project_description: Optional[str],
+) -> str:
+    """Import a csv file into a text classification project."""
+    logger.info(f"Reading csv file: {csv_file}")
+    content_array, categories_array, external_id_array = _read_csv(
+        csv_file, content_column, category_column, external_id_column, csv_separator
+    )
+
+    if project_id is None:
+        if project_title is None:
+            raise ValueError("`project_title` is required if `project_id` is not provided.")
+
+        if project_description is None:
+            raise ValueError("`project_description` is required if `project_id` is not provided.")
+
+        if unique_categories:
+            categories = unique_categories
+        elif categories_array:
+            categories = list(set(categories_array))
+        else:
+            raise ValueError(
+                "Cannot create the project if categories are not provided. You must provide either"
+                " `unique_categories` or `category_column`."
+            )
+
+        project_id = _create_project(kili, project_title, project_description, categories)
+
+        logger.info(f"Created classification project {project_title} with id {project_id}")
+
+    logger.info(f"Adding {len(content_array)} assets to project {project_id}")
+    kili.append_many_to_dataset(
+        project_id=project_id,
+        content_array=content_array,
+        external_id_array=external_id_array,
+        wait_until_availability=True,
+    )
+
+    if category_column is None:
+        return project_id
+
+    assert categories_array is not None, categories_array
+
+    logger.info("Adding labels to project.")
+    json_response_array = _generate_json_response_array(categories_array)
+    kili.append_labels(
+        json_response_array=json_response_array,
+        project_id=project_id,
+        asset_external_id_array=external_id_array,
+        label_type=label_type,
+    )
+
+    return project_id
+
+
+def _read_csv(
+    csv_file: str,
+    content_column: str,
+    category_column: Optional[str],
+    external_id_column: Optional[str],
+    csv_separator: str,
+) -> Tuple[List[str], Optional[List[str]], List[str]]:
+    content_array: List[str] = []
+    categories_array: Optional[List[str]] = [] if category_column is not None else None
+    external_id_array: List[str] = []
+
+    with open(csv_file, newline="", encoding="utf-8") as file:
+        reader = csv.DictReader(file, delimiter=csv_separator)
+
+        for i, row in enumerate(reader):
+            content_array.append(row[content_column])
+
+            if category_column is not None and categories_array is not None:
+                categories_array.append(row[category_column])
```

Comment:
if one row does not have any label (ie no category given), I guess that it will insert `''` in the category array this will create issues while appending labels